compiler/resolution/preFold.cpp

```diff
@@ -390,7 +390,19 @@ static void setRecordDefaultValueFlags(AggregateType* at) {
       ts->addFlag(FLAG_TYPE_NO_DEFAULT_VALUE);
     } else if (isNilableClassType(at)) {
       ts->addFlag(FLAG_TYPE_DEFAULT_VALUE);
-
+    } else if (at->symbol->hasFlag(FLAG_TUPLE)) {
+      Flag flag = FLAG_TYPE_DEFAULT_VALUE;
+      for_fields(field, at) {
+        if (field->isRef()) continue;
+        if (AggregateType* at = toAggregateType(field->getValType())) {
+          setRecordDefaultValueFlags(at);
```

Comment:
This should call isDefaultInitializable instead... but I am working on issue #14854 right now so what you have here might be OK for now if it passes testing and you merge first.

Reply:
Ok. That makes sense, given that `isDefaultInitializable` will call `setRecordDefaultValueFlags` correctly as well.